src/memory_provider.c

```diff
@@ -173,9 +175,41 @@ umf_result_t umfMemoryProviderCreate(const umf_memory_provider_ops_t *ops,
         return UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY;
     }
 
-    assert(ops->version == UMF_VERSION_CURRENT);
-
-    provider->ops = *ops;
+    if (ops->version != UMF_PROVIDER_OPS_VERSION_CURRENT) {
+        LOG_WARN("memory provider ops version \"%d\" is different than current "
+                 "version \"%d\"",
+                 ops->version, UMF_PROVIDER_OPS_VERSION_CURRENT);
+
+        // TODO move to provider_deprecated.c?
+        if (ops->version == UMF_MAKE_VERSION(0, 10)) {
+            umf_memory_provider_ops_10_0_t ops_10_0;
+            memcpy(&ops_10_0, ops, sizeof(ops_10_0));
+
+            provider->ops.version = UMF_PROVIDER_OPS_VERSION_CURRENT;
```

Comment:
Why not leaving it to `0.10`?

Reply:
because after the translation it is no longer a 0.10 struct